### DISCUSSION TITLE
Add Receiver/Emitter apertures to View/Optional rendering

### DISCUSSION THREAD

**Discussion by amrytov** [cyberbotics/webots#5441] [Ideas]
Hi,

Most "directional" nodes like radars or cameras have the corresponding "optional rendering" command which shows rays, light cones and such.   One exception is Emitter/Receiver, which don't have such rendering, although each of them can be made directional by setting their 'aperture' fields.  My request is to add the option to display those apertures.  

The specific occasion:  I'm building an educational scene where children need to program a laser turret to aim the beam at the moving target, simulating sending of commands via the laser channel (this is a space exploration theme, not "star wars").  This is done with a combo of a DistanceSensor which provides a nice laser beam, and an Emitter in the "infrared" mode with a narrow aperture.  The "laser" is good, but having a way to visualize the Emitter would be quite useful, too.

**Comment by omichel** [comment]
That's a good point.
Feel free to explore the source code of Webots to understand how these optional renderings works for other sensors. It should be pretty easy to add some code to support the Emitter and Receiver as well. If you manage to do so, please open a Pull Request that we will be happy to review.

**Reply by amrytov** [reply]
That's a reasonable and fair answer. I quite understand the open source concept "you need a feature - go and implement it yourself", still wish the feature(s) were here already ;-)  The problem is when there is a need, there is usually also a deadline and not much time left to dig into unfamiliar code...

**Reply by omichel** [reply]
If you would like us to do it for you in a quick and efficient way, simply drop me an e-mail at support@cyberbotics.com and I will send you a quote for this.
